modules/tinymce/src/plugins/table/main/ts/actions/InsertTable.ts

```diff
@@ -43,12 +43,17 @@ const insert = (editor: Editor, columns: number, rows: number, colHeaders: numbe
     colGroups: useColumnGroup(editor)
   };
 
-  const table = TableRender.render(rows, columns, rowHeaders, colHeaders, getTableHeaderType(editor), options);
-  Attribute.set(table, 'data-mce-id', '__mce');
+  // Don't create an undo level when inserting the base table HTML otherwise we can end up with 2 undo levels
+  editor.undoManager.ignore(() => {
+    const table = TableRender.render(rows, columns, rowHeaders, colHeaders, getTableHeaderType(editor), options);
+    Attribute.set(table, 'data-mce-id', '__mce');
 
-  const html = Html.getOuter(table);
-  editor.insertContent(html);
+    const html = Html.getOuter(table);
+    editor.insertContent(html);
+    editor.addVisual();
```

Comment:
Why was this added?

Reply:
Because it's needed and was previously being run for the menu item?

Reply:
I should clarify, this adds the visual helpers (dotted border when a table has no borders), etc...